internal/registry/registry.go

```diff
@@ -40,24 +43,24 @@ type RepoPullConfig struct {
 //go:generate mockgen -source=registry.go -package=registry -destination=mock_registry_api.go
 
 type Registry interface {
-	ImageExists(ctx context.Context, image string, po *kmmv1beta1.PullOptions, registryAuthGetter auth.RegistryAuthGetter) (bool, error)
+	ImageExists(ctx context.Context, image string, po *kmmv1beta1.PullOptions, secret *corev1.LocalObjectReference, namespace string) (bool, error)
 	VerifyModuleExists(layer v1.Layer, pathPrefix, kernelVersion, moduleFileName string) bool
-	GetLayersDigests(ctx context.Context, image string, registryAuthGetter auth.RegistryAuthGetter) ([]string, *RepoPullConfig, error)
+	GetLayersDigests(ctx context.Context, image string, secret *corev1.LocalObjectReference, namespace string) ([]string, *RepoPullConfig, error)
 	GetLayerByDigest(digest string, pullConfig *RepoPullConfig) (v1.Layer, error)
 }
 
-type registry struct{}
-
-func NewRegistry() Registry {
-	return &registry{}
+type registry struct {
+	client client.Client
 }
 
-func (r *registry) ImageExists(ctx context.Context, image string, po *kmmv1beta1.PullOptions, registryAuthGetter auth.RegistryAuthGetter) (bool, error) {
-	pullConfig, err := r.getPullOptions(ctx, image, po, registryAuthGetter)
-	if err != nil {
-		return false, fmt.Errorf("failed to get pull options for image %s: %w", image, err)
+func NewRegistry(client client.Client) Registry {
+	return &registry{
+		client: client,
 	}
-	_, err = r.getImageManifest(ctx, image, pullConfig)
+}
+
+func (r *registry) ImageExists(ctx context.Context, image string, po *kmmv1beta1.PullOptions, secret *corev1.LocalObjectReference, namespace string) (bool, error) {
```

Comment:
That means that if we add another `RegistryAuthGetter` in the future that doesn't relies on a secret we will have to change that function instead of just passing another `RegistryAuthGetter` as an argument.